src/System.IO.FileSystem/src/System/IO/MultiplexingWin32WinRTFileSystem.cs

```diff
@@ -145,33 +147,63 @@ private bool ShouldUseWinRT(string fullPath, bool isCreate)
-                Interop.WIN32_FIND_DATA findData = new Interop.WIN32_FIND_DATA();
-                using (SafeFindHandle handle = Interop.mincore.FindFirstFile(fullPath, ref findData))
-                {
-                    int error = Marshal.GetLastWin32Error();
+                int error = Interop.ERROR_SUCCESS;
 
-                    if (handle.IsInvalid)
+                // first use GetFileAttributesEx as it is faster than FindFirstFile and requires minimum permissions
+                Interop.WIN32_FILE_ATTRIBUTE_DATA data = new Interop.WIN32_FILE_ATTRIBUTE_DATA();
+                if (Interop.mincore.GetFileAttributesEx(fullPath, Interop.GET_FILEEX_INFO_LEVELS.GetFileExInfoStandard, ref data))
+                {
+                    // got the attributes
+                    if ((data.fileAttributes & Interop.FILE_ATTRIBUTE_DIRECTORY) != 0 ||
+                        (data.fileAttributes & Interop.FILE_ATTRIBUTE_REPARSE_POINT) == 0)
                     {
-                        if (error == Interop.ERROR_ACCESS_DENIED)
-                        {
-                            // The path was not accessible with Win32, so try WinRT
-                            useWinRt = true;
-                            break;
-                        }
-                        else if (error != Interop.ERROR_PATH_NOT_FOUND && error != Interop.ERROR_FILE_NOT_FOUND)
-                        {
-                            // We hit some error other than ACCESS_DENIED or NOT_FOUND,
-                            // Default to Win32 to provide most accurate error behavior
-                            break;
-                        }
+                        // we have a directory or a file that is not a reparse point
+                        // useWinRt = false;
+                        break;
                     }
                     else
                     {
-                        // Use WinRT for placeholder files
-                        useWinRt = IsPlaceholderFile(findData);
-                        break;
+                        // we need to get the find data to determine if it is a placeholder file
+                        Interop.WIN32_FIND_DATA findData = new Interop.WIN32_FIND_DATA();
+                        using (SafeFindHandle handle = Interop.mincore.FindFirstFile(fullPath, ref findData))
+                        {
+                            if (!handle.IsInvalid)
+                            {
+                                // got the find data, use WinRT for placeholder files
+
+                                Debug.Assert((findData.dwFileAttributes & Interop.FILE_ATTRIBUTE_DIRECTORY) == 0);
+                                Debug.Assert((findData.dwFileAttributes & Interop.FILE_ATTRIBUTE_REPARSE_POINT) != 0);
+
+                                useWinRt = findData.dwReserved0 == Interop.IO_REPARSE_TAG_FILE_PLACEHOLDER;
+                                break;
+                            }
+                            else
```

Comment:
Sure, I considered this initially but I'm sensitive to folks making inadvertent changes that blow away GLE.  I suppose the Assert should help protect us here.